### PR TITLE
troubleshoot weak ssl certificates used by web sockets

### PR DESCRIPTION
**Pull request by MrRagga-**

### CODE REVIEW

File: _includes/manuals/1.8/7.1_novnc.md [hunk 1]
```diff
@@ -30,3 +30,6 @@ For VNC only, encrypted connections are the default on new installations.  If yo
 * Try a telnet/netcat connection from the Foreman host to the hypervisor hostname/port
 * The penultimate argument of websockify.py is the listening port number, check if your web client host can telnet to it
 * If using Firefox, check the known issues above and set the config appropriately
+* WebSock error: [object Event] can be caused by a self signed certificate, where the certificate`s algoorithm is too weak, e.g. SHA1. Debugging with e.g. the Firefox Java Script Console will print a warning similar to "This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1. [https://developer.mozilla.org/docs/Security/Weak_Signature_Algorithm]". Check your current algorithm used fot the SSL certificate with openssl:
+  # openssl x509 -in /etc/ssl/certs/foreman.domain.tld.crt -text -noout |grep Algorithm
```
Comment (dLobatog) [inline _includes/manuals/1.8/7.1_novnc.md]: This `#` is currently interpreted as an h1 in markdown, make sure it's indented (and escaped if needed properly).

File: _includes/manuals/1.8/7.1_novnc.md [hunk 2]
```diff
@@ -30,3 +30,6 @@ For VNC only, encrypted connections are the default on new installations.  If yo
 * Try a telnet/netcat connection from the Foreman host to the hypervisor hostname/port
 * The penultimate argument of websockify.py is the listening port number, check if your web client host can telnet to it
 * If using Firefox, check the known issues above and set the config appropriately
+* WebSock error: [object Event] can be caused by a self signed certificate, where the certificate`s algoorithm is too weak, e.g. SHA1. Debugging with e.g. the Firefox Java Script Console will print a warning similar to "This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1. [https://developer.mozilla.org/docs/Security/Weak_Signature_Algorithm]". Check your current algorithm used fot the SSL certificate with openssl:
```
Comment (dLobatog) [inline _includes/manuals/1.8/7.1_novnc.md]: `used fot` -> `used for` 
`algoorithm` -> `algorithm`

**Comment by dLobatog** [comment]
@MrRagga- Thanks! I've seen this issue before, however what I see in the pull request is an explanation of what is going on, but no way to fix it?

**Comment by dLobatog** [comment]
@MrRagga- ping?

**Comment by MrRagga-** [comment]
Using stronger SHA-2 algorithm's seems to work.

File: _includes/manuals/1.8/7.1_novnc.md (hunk 2 above)
Comment (dLobatog) [inline _includes/manuals/1.8/7.1_novnc.md]: `Java Script` -> `JavaScript`

**Comment by dLobatog** [comment]
@MrRagga- Thanks, could you add that to the doc & update the formatting issues I've commented inline? I'd be glad to merge after that.

Notice you should change the 1.9 pages of the manual now, not 1.8. :smile:

**Comment by dLobatog** [comment]
@MrRagga- Any updates?

**Comment by MrRagga-** [comment]
see commit from 19th of august? https://github.com/MrRagga-/theforeman.org/commit/714d6bee4a873c1c9bc192ecb3822588b975217f

File: _includes/manuals/1.9/7.1_novnc.md [hunk 3]
```diff
@@ -30,3 +30,7 @@ For VNC only, encrypted connections are the default on new installations.  If yo
 * Try a telnet/netcat connection from the Foreman host to the hypervisor hostname/port
 * The penultimate argument of websockify.py is the listening port number, check if your web client host can telnet to it
 * If using Firefox, check the known issues above and set the config appropriately
+* *WebSock error: [object Event]* can be caused by a self signed certificate, where the certificate`s algorithm is too weak, e.g. SHA1. Debugging with e.g. the Firefox JavaScript Console will print a warning similar to "This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1. [https://developer.mozilla.org/docs/Security/Weak_Signature_Algorithm]". To solve this issue, use SHA-2 algorithms instead.
+Check your current algorithm used for the SSL certificate with openssl:
+  $ openssl x509 -in /etc/ssl/certs/foreman.domain.tld.crt -text -noout |grep Algorithm
```
Comment (dLobatog) [inline _includes/manuals/1.9/7.1_novnc.md]: I think these two lines need another blank line in between and more indentation to show up formatted like code.

File: _includes/manuals/1.9/7.1_novnc.md [hunk 4]
```diff
@@ -30,3 +30,7 @@ For VNC only, encrypted connections are the default on new installations.  If yo
 * Try a telnet/netcat connection from the Foreman host to the hypervisor hostname/port
 * The penultimate argument of websockify.py is the listening port number, check if your web client host can telnet to it
 * If using Firefox, check the known issues above and set the config appropriately
+* *WebSock error: [object Event]* can be caused by a self signed certificate, where the certificate`s algorithm is too weak, e.g. SHA1. Debugging with e.g. the Firefox JavaScript Console will print a warning similar to "This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1. [https://developer.mozilla.org/docs/Security/Weak_Signature_Algorithm]". To solve this issue, use SHA-2 algorithms instead.
```
Comment (dLobatog) [inline _includes/manuals/1.9/7.1_novnc.md]: Maybe find a text to link the Mozilla docs article?

**Comment by dLobatog** [comment]
@MrRagga- Thanks, I didn't realize there was a change on 1.9. Can you remove the change to the 1.8 manual & address the comments inline? Almost there :smile: !

**Comment by mmoll** [comment]
what's the status here?

**Comment by MrRagga-** [comment]
Sorry for being that late. Hopefully it fits now.

**Comment by dLobatog** [comment]
@MrRagga- Thanks! It seems like the quotes problem is still there so it's formatted wrong. I'll change it on merge and put the changes in 1.10 too, which is the most current version.

**Comment by dLobatog** [comment]
@MrRagga- I rebased your commits & moved the changes to 1.10 - merged as f23f3a6d701b5fa22a74f3f2612cf75452c94534 , thanks!

**Pull request closed without merging**